tracer/tracer.go

```diff
@@ -133,6 +136,42 @@ func (t *Tracer) SetServiceInfo(name, app, appType string) {
 	}
 }
 
+// SetMeta adds an arbitrary meta field at the tracer level.
+// This will append those tags to each span created by the tracer.
+func (t *Tracer) SetMeta(key, value string) {
+	if t == nil { // Defensive, span could be initialized with nil tracer
+		return
+	}
+
+	t.metaMu.Lock()
+	if t.meta == nil {
+		t.meta = make(map[string]string)
+	}
+	t.meta[key] = value
+	t.metaMu.Unlock()
+}
+
+// getAllMeta returns all the meta set by this tracer.
+// In most cases, it is nil.
+func (t *Tracer) getAllMeta() map[string]string {
```

Comment:
The reason it's here is locking, this way we use `RLock` within the tracer code, the span code does not have to worry about it. Should we not do this I think span code would have to deal internally with the private mutex. This is perfectly doable, but I tend to prefer having objects dealing with their mutex internally, and not require others to care about it. 

Additionnally, Python code uses https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/tracer.py#L164 which calls https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/span.py#L145 so it also has this deep copy, only it's done in span.py instead of tracer.py.

I can move the code to span.go if you want to, just exposing the reason it's here. Note that getAllMeta is private, it's not part of the API anyway.